File: common/security/src/main/java/com/example/chat/auth/core/util/AuthConverter.java
package com.example.chat.auth.core.util;

import com.example.chat.auth.core.model.AuthenticatedUser;
import org.springframework.security.oauth2.jwt.Jwt;
import org.springframework.stereotype.Component;

/**
 * JWT → AuthenticatedUser 변환기
 * {@link com.example.chat.auth.core.annotation.CurrentUser} 어노테이션에서 사용
 */
@Component("authConverter")
public class AuthConverter {

    public AuthenticatedUser convert(Jwt jwt) {
        return AuthenticatedUser.from(jwt);
    }
}

File: common/security/src/main/kotlin/com/example/chat/auth/core/util/AuthConverter.kt

package com.example.chat.auth.core.util

import com.example.chat.auth.core.model.AuthenticatedUser
import org.springframework.security.oauth2.jwt.Jwt
import org.springframework.stereotype.Component

/**
 * JWT를 AuthenticatedUser로 변환하는 컨버터
 * @CurrentUser 어노테이션에서 사용됨
 */
@Component("authConverter")
class AuthConverter {
    fun convert(jwt: Jwt?): AuthenticatedUser? {
        return AuthenticatedUser.from(jwt)
    }
}
